Src/AutoFixtureUnitTest/MailAddressGeneratorTests.cs

```diff
@@ -87,13 +94,39 @@ public void CreateWithMailAddressRequestReturnsCorrectResultUsingLocalPartFromCo
         public void CreateWithMailAddressRequestReturnsNoSpecimenWhenContextReturnsNullLocalPart()
         {
             // Fixture setup
-            var request = typeof(MailAddress);         
+            var request = typeof(MailAddress);
+            var anonymousDomainName = new DomainName(Guid.NewGuid().ToString());
+
+            var context = new DelegatingSpecimenContext()
+            {
+                OnResolve = r =>
+                {
+                    Assert.True(typeof(EmailAddressLocalPart).Equals(r) || typeof(DomainName).Equals(r));
+                    return typeof(DomainName).Equals(r) ? anonymousDomainName : null;
+                }
+            };
+            var sut = new MailAddressGenerator();
+            // Exercise system
+            var expectedResult = new NoSpecimen(request);

```

Comment:
Technically, there's no difference :smile: 

FWIW, I find the related heuristics described in [Code Complete](http://amzn.to/1xioCmw) reasonable: introduce variables as close to where they're used as possible. That makes the code easier to read, because the distance between introduction and use is as short as possible.